src/auto-evo/mutations/CommonMutationFunctions.cs

```diff
@@ -53,6 +55,46 @@ public enum Direction
         Rear,
     }
 
+    public static MicrobeSpecies GenerateRandomSpecies(MicrobeSpecies mutated, MutationWorkMemory workMemory,
+        Random random, float mp = 300)
+    {
+        var mutationStrategy = new AddOrganelleAnywhere(_ => true);
+
+        GameWorld.SetInitialSpeciesProperties(mutated, workMemory.WorkingMemory1, workMemory.WorkingMemory2);
+
+        while (mp > 0)
+        {
+            var mutationList = mutationStrategy.MutationsOf(mutated, mp, true)?.OrderBy(_ => random.Next()).ToList();
```

Comment:
Wouldn't FirstOrDefault be better here? That avoids potentially building a list of 100 items but then discarding all but the first one.